pythia/scripts/features/extract_features_vmb.py

```diff
@@ -0,0 +1,179 @@
+# Requires maskrcnn-benchmark to be built and installed
+import argparse
+import glob
+import os
+
+import cv2
+import numpy as np
+import torch
+from PIL import Image
+
+from maskrcnn_benchmark.config import cfg
+from maskrcnn_benchmark.layers import nms
+from maskrcnn_benchmark.modeling.detector import build_detection_model
+from maskrcnn_benchmark.structures.image_list import to_image_list
+from maskrcnn_benchmark.utils.model_serialization import load_state_dict
+from pythia.utils.general import download_file
+
+
+class FeatureExtractor:
+    MODEL_URL = (
+        "https://dl.fbaipublicfiles.com/pythia/detectron_model/detectron_model.pth"
+    )
+    CONFIG_URL = (
+        "https://dl.fbaipublicfiles.com/pythia/detectron_model/detectron_model.yaml"
+    )
+
+    def __init__(self):
+        self.args = self.get_parser().parse_args()
+        self.detection_model = self._build_detection_model()
+
+        if not os.path.exists(self.args.output_folder):
+            os.makedirs(self.args.output_folder, exist_ok=True)
+
+    def _try_downloading_necessities(self):
+        if self.args.model_file is None:
+            print("Downloading model and configuration")
+            self.args.model_file = "./detectron_model.pth"
+            self.args.config_file = "./detectron_model.yaml"
+            download_file(self.MODEL_URL)
+            download_file(self.CONFIG_URL)
+
+    def get_parser(self):
+        parser = argparse.ArgumentParser()
+        parser.add_argument(
+            "--model_file", default=None, type=str, help="Detectron model file"
+        )
+        parser.add_argument(
+            "--config_file", default=None, type=str, help="Detectron config file"
+        )
+        parser.add_argument("--batch_size", type=int, default=2, help="Batch size")
+        parser.add_argument(
+            "--output_folder", type=str, default="./output", help="Output folder"
+        )
+        parser.add_argument("--image_dir", type=str, help="Image directory or file")
+
+        return parser
+
+    def _build_detection_model(self):
+        cfg.merge_from_file(self.args.config_file)
+        cfg.freeze()
+
+        model = build_detection_model(cfg)
+        checkpoint = torch.load(self.args.model_file, map_location=torch.device("cpu"))
+
+        load_state_dict(model, checkpoint.pop("model"))
+
+        model.to("cuda")
+        model.eval()
+        return model
+
+    def _image_transform(self, path):
+        img = Image.open(path)
+        im = np.array(img).astype(np.float32)
+        im = im[:, :, ::-1]
+        im -= np.array([102.9801, 115.9465, 122.7717])
+        im_shape = im.shape
+        im_size_min = np.min(im_shape[0:2])
+        im_size_max = np.max(im_shape[0:2])
+        im_scale = float(800) / float(im_size_min)
+        # Prevent the biggest axis from being more than max_size
+        if np.round(im_scale * im_size_max) > 1333:
+            im_scale = float(1333) / float(im_size_max)
+        im = cv2.resize(
```

Comment:
Maybe use a torch transform here, or if this is consistent with transforms elsewhere ignore my comment.

Reply:
The underlying engine in torch transforms is PIL and the C2 transforms are cv2 based. To keep features exactly the same, it'd be better to use the same image processing engine.